plugin/Cache/CacheHandler.php

```diff
@@ -68,9 +68,31 @@ public function clear()
 
 	public function clean()
 	{
-		foreach ($this->_pSDKWrapper->getCache() as $pCache) {
-			/* @var $pCache onOfficeSDKCache */
-			$pCache->cleanup();
+		if(!empty($this->setUpApiCallWithCurlOptions())){
+			foreach ($this->_pSDKWrapper->getCache() as $pCache) {
+				/* @var $pCache onOfficeSDKCache */
+				$pCache->cleanup();
+			}
 		}
 	}
+
+
+	/**
+	 *
+	 */
+
+	public function setUpApiCallWithCurlOptions(): array
```

Comment:
The method is kind of awkward to test. I think that is because this method does an entirely different thing: it tests for the availability of the API. This has nothing to do with the CacheHandler, except that CacheHandler is going to call it. 

=> Please create a new class (for example `APIAvailabilityChecker`) and move the method there. The method should get a new name as well, for example `checkAvailability()`.

Reply:
In which folder should I create new class?